src/core/fieldformatter/qgsvaluerelationfieldformatter.cpp

```diff
@@ -162,3 +172,58 @@ QStringList QgsValueRelationFieldFormatter::valueToStringList( const QVariant &v
   }
   return checkList;
 }
+
+
+QSet<QString> QgsValueRelationFieldFormatter::expressionFormVariables( const QString &expression )
+{
+  const QStringList formVariables( QgsExpressionContextUtils::formScope()->variableNames() );
+  QSet<QString> variables;
+
+  for ( auto const &variable : formVariables )
```

Comment:
Instead of this, I'd go:
```
   std::unique_ptr< QgsExpressionContextScope > scope(QgsExpressionContextUtils::formScope() );
   QSet< QString > formVariables = scope->variableNames().toSet();
   const QSet< QString > usedVariables = QgsExpression( expression ).referencedVariables();
   formVariables.intersect( usedVariables );
   return formVariables;
```